packages/blade/src/components/Dropdown/_decisions/decisions.md

```diff
@@ -0,0 +1,672 @@
+        <Button>Primary Button</Button>
+      </>
+    }
+  />
+</ActionList>
+```
+
+<details>
+<summary>Alternate Prop-based API</summary>
+
+**Props and constrained**
+
+```jsx
+<ActionList backgroundLevel={2}>
+  <ActionList.Header title="Recent Searches" icon={HistoryIcon} />
+  <ActionList.SectionHeading title="Hello" />
+  <ActionList.Item
+    title="Item Value"
+    desciption="Item Description"
+    leading={StarIcon} // Can be IconComponent type or Image URL
+    trailing="⌘ + Space"
+  />
+  <ActionList.Divider />
+  <ActionList.Footer
+    title="Footer Title"
+    description="Footer Description"
+    leading={DocIcon}
+    trailing={{
+      primary: {
+        onClick: () => {},
+        text: 'Primary Action',
+      },
+    }}
+    // OR
+    trailing={ArrowRightIcon}
+    // OR
+    trailing={{
+      primary: {
+        onClick: () => {
+          console.log('Feedback Yes Clicked');
+        },
+        icon: CheckIcon,
+      },
+      secondary: {
+        onClick: () => {
+          console.log('Feedback No Clicked');
+        },
+        icon: CrossIcon,
+      },
+    }}
+  />
+</ActionList>
+```
+
+</details>
+
+### `ActionList` Parent Component
+
+Wrapper for all ActionList compound components. Should handle the logic of selections.
+
+#### API
+
+Usage
+
+```jsx
+<ActionList>{/* Child components */}</ActionList>
+```
+
+Props
+
+```ts
+type ActionListProps = {
+  children: React.ReactNode;
+
+  /**
+   * Renders simple menu or checkboxes depending on the variant
+   *
+   * @default single
+   */
+  variant?: 'single' | 'multiple';
+
+  /** Callback that gets called when item is checked/clicked */
+  onChange?: () => void;
+
+  /**
+   * Changes the background surface level color
+   *
+   * @default 2
+   */
+  backgroundLevel: 2 | 3;
+};
+```
+
+### `ActionList.Item`
+
+![](2022-11-29-17-29-01.png)
+
+#### Design
+
+- [Figma - ActionListItem / Body](https://www.figma.com/file/sAdplk2uYnI2ILnDKUxycW/Blade---Banking-Dark?node-id=13136%3A378703&t=RoEboUS3qWdStKK6-0)
+
+#### API
+
+Usage
+
+```jsx
+<ActionList.Item
+  title="Accept API"
+  value="accept"
+  desciption="Accept this API"
+  leading={<StarIcon />}
+  intent="negative"
+  trailing={<Text>⌘ + Space + A</Text>}
+  onClick={() => {
+    console.log('Item clicked');
+  }}
+/>
+```
+
+Props
+
+```ts
+type ActionListItemProps = {
+  title: string;
+
+  description: string;
+
+  /** Value that we use in JS form. Similar to `value` in `Checkbox` */
+  value: string;
+
+  /**
+   * Leading Icon or assets. Accepts JSX Component
+   */
+  leading?: React.ReactNode;
+
+  /**
+   * Trailing Assets. Should ideally be Text component for now
+   */
+  trailing?: React.ReactNode;
+
+  /**
+   * Changes item color to red
+   *
+   * @default neutral
+   */
+  intent?: 'negative' | 'neutral';
+
+  /** Click event on item */
+  onClick?: () => void;
+
+  /** URL. Should render item as `a` when this is passed */
+  href?: string;
```

Comment:
ActionList can be used in navbars also so it can take users to some different links. Or there could be some action which takes people to different links.

Reply:
how will this behave when `href` and `onClick` both are passed?

Reply:
like an anchor tag with onClick event on it. `href` is the deciding factor. If it is present then its `a` or its `buttont`

Reply:
i think we shall confirm with design once for this because they haven't created a link actionlist item which might look different than our regular link. so if they haven't then maybe we shall add this prop later once they add